src/linear_algebra/tensor_product.lean

```diff
@@ -756,6 +760,17 @@ lemma lift_comp_map (i : P →ₗ[R] Q →ₗ[R] Q') (f : M →ₗ[R] P) (g : N
   (lift i).comp (map f g) = lift ((i.comp f).compl₂ g) :=
 ext $ λ _ _, by simp only [lift.tmul, map_tmul, linear_map.compl₂_apply, linear_map.comp_apply]
 
+@[simp] lemma map_id : map (id : M →ₗ[R] M) (id : N →ₗ[R] N) = id :=
+by { ext, simp only [mk_apply, id_coe, compr₂_apply, id.def, map_tmul], }
+
+@[simp] lemma map_pow (f : M →ₗ[R] M) (g : N →ₗ[R] N) (n : ℕ) :
+  (map f g)^n = map (f^n) (g^n) :=
+begin
+  induction n with n ih,
+  { simp only [pow_zero, one_eq_id, map_id], },
+  { simp only [pow_succ', ih, mul_eq_comp, map_comp], },
+end
```

Comment:
If you're going to add this lemma, I'd probably also add `map_mul : map (f₂ * f₁) (g₂ * g₁) = (map f₂ g₂) * (map f₁ g₁)` and `map_one : map (1 : M →ₗ[R] M) (1 : N →ₗ[R] N) = 1`, which you can then use in the proof.

Reply:
Hmm, there is already `map_comp`. I'll have to think for a moment to decide what should be the canonical form etc.

Reply:
(but basically I agree)

Reply:
I don't think there is a right choice of canonical form here, just as there isn't for `perm`. I think `map_mul : _ := map_comp` is a totally reasonable way to handle this.